taskwiki/main.py

```diff
@@ -123,6 +132,34 @@ def info(self):
                 util.show_in_split(out, name='info', activate_cursorline=True)
             break  # Show only one task
 
+    @errors.pretty_exception_handler
+    def open(self):
+        compatible_annotation_found = False
+        for vimwikitask in self.tasks:
+            for annotation in vimwikitask.task["annotations"]:
+                annotation = annotation["description"]
+                proc = subprocess.Popen(["xdg-open", annotation])
+                try:
+                    proc.wait(0.3)
+                except subprocess.TimeoutExpired:
+                    compatible_annotation_found = True
+
+        if not compatible_annotation_found:
+            print("No compatible annotation found.")
+
+    @errors.pretty_exception_handler
+    def note(self):
+        for vimwikitask in self.tasks:
+            if self.taskopen_notes_regex not in [
+                a["description"] for a in vimwikitask.task["annotations"]
+            ]:
+                self.annotate(self.taskopen_notes_regex)
+
+            note_path = os.path.join(self.taskopen_notes_folder,
+                                     vimwikitask.task["uuid"] + ".md")
```

Comment:
taskopen actually uses `.txt` by default (it's configurable, of course). In taskwiki, the most natural thing to do would be to use the same format as used by the current vimwiki.

Maybe it also makes sense to let vimwiki open the link ensuring that the directory is created and so on: https://github.com/vimwiki/vimwiki/blob/619f04f89861c58e5a6415a4f83847752928252d/autoload/vimwiki/base.vim#L838? Not sure if that makes sense unless taskopen_notes_folder is inside a wiki, but I'll certainly configure it that way.

Reply:
The new approach uses taskopen itself, so this will not be an issue.